torch/serialization.py

```diff
@@ -760,7 +773,27 @@ def load(
                                   " silence this warning)", UserWarning)
                     opened_file.seek(orig_position)
                     return torch.jit.load(opened_file, map_location=map_location)
+                if pickle_module is None:
+                    try:
+                        return _load(opened_zipfile, map_location, _weights_only_unpickler, **pickle_load_args)
+                    except RuntimeError:
+                        # Rethrow the exception if only safe unpickler is allowed
+                        if weight_load_only:
+                            raise
+                        warnings.warn(UNSAFE_MESSAGE)
```

Comment:
so this would generate an error message always if people legitimately want to pickle pytorch models (our tutorials still mention it as an option). Maybe make `weight_load_only` to be 3-way, i.e. default None? If it's None - generate the warning, but if it's explicitly passed as False - not?

also, should we in future releases switch default to be safe only?

Reply:
See @ezyang comment below: we propose we already change the `weights_only` default to True, this way I think semantic would be much simpler:
 - True - uses JIT or safe unpickler
 - False - uses default pickler or any pickle module that is passed as an argument